consensus/common/src/main/java/tech/pegasys/pantheon/consensus/common/jsonrpc/AbstractGetSignerMetricsMethod.java

```diff
@@ -0,0 +1,123 @@
+        parameters.optional(request.getParams(), 1, BlockParameter.class);
+
+    final long fromBlockNumber = getFromBlockNumber(startBlockParameter);
+    final long toBlockNumber = getEndBlockNumber(endBlockParameter);
+
+    if (isValidParameters(fromBlockNumber, toBlockNumber)) {
+
+      final Map<Address, SignerMetricResult> proposersMap = new HashMap<>();
+      final long lastBlockIndex = toBlockNumber - 1;
+
+      // go through each block (startBlock is inclusive and endBlock is exclusive)
+      LongStream.range(fromBlockNumber, toBlockNumber)
+          .forEach(
+              currentIndex -> {
+                Optional<BlockHeader> blockHeaderByNumber =
+                    blockchainQueries.getBlockHeaderByNumber(currentIndex);
+
+                // Get the number of blocks from each proposer in a given block range.
+                blockHeaderByNumber.ifPresent(
+                    header -> {
+                      final Address proposerAddress = blockInterface.getProposerOfBlock(header);
+                      SignerMetricResult signerMetric =
+                          proposersMap.computeIfAbsent(proposerAddress, SignerMetricResult::new);
+                      signerMetric.incrementeNbBlock();
+                      // Add the block number of the last block proposed by each validator
+                      signerMetric.setLastProposedBlockNumber(currentIndex);
+
+                      // Get All validators present in the last block of the range even
+                      // if they didn’t propose a block
+                      if (currentIndex == lastBlockIndex) {
+                        blockInterface
+                            .validatorsInBlock(header)
+                            .forEach(
+                                address ->
+                                    proposersMap.computeIfAbsent(
+                                        proposerAddress, SignerMetricResult::new));
+                      }
+                    });
+              });
+
+      return new JsonRpcSuccessResponse(request.getId(), new ArrayList<>(proposersMap.values()));
+    } else {
+      return new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
+    }
+  }
+
+  private long getFromBlockNumber(final Optional<BlockParameter> startBlockParameter) {
+    return startBlockParameter
+        .map(
+            blockParameter ->
+                blockParameter
+                    .getNumber()
+                    .orElse(blockchainQueries.headBlockNumber() - DEFAULT_RANGE_BLOCK))
+        .orElseGet(() -> blockchainQueries.headBlockNumber() - DEFAULT_RANGE_BLOCK);
+  }
+
+  private long getEndBlockNumber(final Optional<BlockParameter> endBlockParameter) {
+    return endBlockParameter
+        .map(
+            blockParameter ->
+                blockParameter.getNumber().orElse(blockchainQueries.headBlockNumber()))
+        .orElseGet(blockchainQueries::headBlockNumber);
```

Comment:
`BlockParameter` is slightly more annoying than you might imagine.  It's `getNumber` method will return empty if any of the "alias" options were used (`earliest`, `latest` or `pending`).

You'll need something like:
```
private long resolveBlockNumber(final BlockParameter param) {
   if (param.getNumber().isPresent()) {
       return param.getNumber().getAsLong();
   if (param.isEarliest) {
        return BlockHeader.GENESIS_BLOCK_NUMBER;
   } else if (param.isLatest() || param.isPending()) {
        return blockchain.getChainHeadBlockNumber();
   } else {
        throw new IllegalStateException("Unknown block parameter type.");
    }
  }
```